jitbuilder/release/src/Worklist.cpp

```diff
@@ -0,0 +1,343 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2016, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+
+#include <iostream>
+#include <stdlib.h>
+#include <stdint.h>
+#include <dlfcn.h>
+#include <errno.h>
+
+#include "Jit.hpp"
+#include "ilgen/TypeDictionary.hpp"
+#include "ilgen/MethodBuilder.hpp"
+#include "ilgen/BytecodeBuilder.hpp"
+#include "ilgen/VirtualMachineState.hpp"
+#include "Worklist.hpp"
+
+using std::cout;
+using std::cerr;
+
+#define REPORT_BCI(bci,e_bci)   { if ((bci) != (e_bci)) { cerr << "Error: bytecode " << (bci) << " not found in worklist ( expected" << (e_bci) << ")\n"; return false; } }
+
+#define REPORT_VMSTATE(b,s)
+
+//#define REPORT_VMSTATE(b,s)     { if ((b)->vmState() != (s)) { cerr << "Error: bytecode " << ((b)->bcIndex()) << " does not have correct vmState (" << (s) << ")\n"; return false; } }
+
+#define REPORT_RESULT(p,r,e)    { if ((r) == (e)) { success++; } else { cerr << "Error: path " << (p) << " generated wrong result (" << (r) << " versus " << (e) << ")\n"; fails++; } }
+
+
+static bool verbose = false;
+static int32_t fails = 0;
+static int32_t success = 0;
+
+// BC0: r = p;
+// BC1: r++;
+//      if (r > 10) goto BC3;
+// BC2: if (r > 5)  goto BC5;
+// BC4: if (r < 3)  goto BC1; // loop back
+// BC9: return 9;
+//
+// BC5: if (r < 8) goto BC11  (return 11)
+// BC17: return 17;
+//
+// BC7: if (r < 17) goto BC1: // loop back
+// BC8: return 8;
+//
+//
+// BC3: if (r > 15) goto BC7  (return 7)
```

Comment:
I do not believe this line would return the value 7.  Or does the (return 7) just signify the goto BC7?